test/swe2d/test_rossby_wave.py

```diff
@@ -0,0 +1,342 @@
+    φ = 0.771*(B/cosh(B*ξ))**2
+    dφdx = -2*B*φ*tanh(B*ξ)
+    C = -0.395*B*B
+
+    # Zeroth order terms
+    eta_terms = φ*0.25*(3 + 6*y*y)*Ψ
+    if order == 0:
+        return interpolate(eta_terms, H_2d)
+
+    # Unnormalised Hermite series coefficients for eta
+    eta = np.zeros(28)
+    eta[0]  = -3.0714300e+00
+    eta[2]  = -0.3508384e-01
+    eta[4]  = -0.1861060e-01
+    eta[6]  = -0.2496364e-03
+    eta[8]  =  0.1639537e-04
+    eta[10] = -0.4410177e-06
+    eta[12] =  0.8354759e-09
+    eta[14] = -0.1254222e-09
+    eta[16] =  0.1573519e-11
+    eta[18] = -0.1702300e-13
+    eta[20] =  0.1621976e-15
+    eta[22] = -0.1382304e-17
+    eta[24] =  0.1066277e-19
+    eta[26] = -0.1178252e-21
+
+    # Hermite polynomials
+    polynomials = [Constant(1.0), 2*y]
+    for i in range(2, 28):
+        polynomials.append(2*y*polynomials[i-1] - 2*(i-1)*polynomials[i-2])
+
+    # First order terms
+    eta_terms += C*φ*0.5625*(-5 + 2*y*y)*Ψ
+    eta_terms += φ*φ*Ψ*sum(eta[i]*polynomials[i] for i in range(28))
+
+    return interpolate(eta_terms, H_2d)
+
+
+def run(refinement_level, reference_refinement_level=384, **model_options):
+    print_output("--- running refinement level {:}".format(refinement_level))
+    order = model_options.pop('expansion_order')
+
+    # Set up domain
+    lx, ly = 48, 24
+    nx, ny = 2*refinement_level, refinement_level
+    params = {'partition': True, 'overlap_type': (DistributedMeshOverlapType.VERTEX, 10)}
+    mesh2d = PeriodicRectangleMesh(nx, ny, lx, ly, distribution_parameters=params)
+    x, y = SpatialCoordinate(mesh2d)
+    mesh2d.coordinates.interpolate(as_vector([x-lx/2, y-ly/2]))
+    T = 120.0
+
+    physical_constants['g_grav'].assign(1.0)
+
+    # Create solver object
+    solver_obj = solver2d.FlowSolver2d(mesh2d, Constant(1.0))
+    options = solver_obj.options
+    options.timestepper_type = 'CrankNicolson'
+    options.timestep = 0.96/refinement_level
+    options.simulation_export_time = 5.0
+    options.simulation_end_time = T
+    options.use_grad_div_viscosity_term = False
+    options.use_grad_depth_viscosity_term = False
+    options.horizontal_viscosity = None
+    options.no_exports = True and refinement_level <= 48
+    solver_obj.create_function_spaces()
+    options.coriolis_frequency = interpolate(y, solver_obj.function_spaces.P1_2d)
+    options.update(model_options)
+
+    # Apply boundary conditions
+    for tag in mesh2d.exterior_facets.unique_markers:
+        solver_obj.bnd_functions['shallow_water'][tag] = {'uv': Constant(as_vector([0., 0.]))}
+
+    # Apply initial conditions (asymptotic solution at initial time)
+    uv_a = asymptotic_expansion_uv(solver_obj.function_spaces.U_2d, order=order)
+    elev_a = asymptotic_expansion_elev(solver_obj.function_spaces.H_2d, order=order)
+    solver_obj.assign_initial_conditions(uv=uv_a, elev=elev_a)
+
+    # Solve PDE and save to HDF5
+    solver_obj.iterate()
+    fname = "{:d}_{:d}".format(refinement_level, mesh2d.comm.size)
```

Comment:
So the reason I use HDF5 checkpointing is because I got to the end of one run, projected the solution onto the reference mesh and then just got `Killed`, probably because the reference space was pushing my memory limits. I just didn't want to lose the solution again.

But I agree that it is a bit of a weird thing to do in a test. Will modify now.